src/common/debug.h

```diff
@@ -20,13 +20,13 @@
 /* Global version of the stuff in common/dout.h
  */
 
-#define dout(v) ldout((dout_context), v)
+#define dout(v) ldout((dout_context), (v))
```

Comment:
what is the advantage of doing so? under what circumstances, `v` can be interpreted in a wrong way?

Reply:
We add some linux-style rate-limit check in the dout implementation to limit the amount of log can be produced. e,g.:
```c++
#define dout_impl(cct, sub, v)						\
  do {									\
  if (cct->_conf->subsys.should_gather(sub, v)) {			\
    static DEFINE_RATELIMIT_STATE(__rs);                                \
    if (v < 1 && ratelimit_check(&__rs)) { break; }                     \
```
It turned out it could not work well sometimes when the input was something like:
```c++
dout(read_only ? 5 : 0)
```

Of course we could fix the above problem by either:
```c++
 if ((v) < 1 && ratelimit_check(&__rs)) { break; }  
```
or 
```
int log_level = read_only ? 5 : 0;
dout(log_level)
```
which also reminds me protection of this kind should be a more generally way to avoid issues like this one and hence makes sense to me...